Feature-task5:

AlexLukashevich95 opened the pull request:
-added hibernate

AlexLukashevich95 committed:
Feature-task5:
-added hibernate

Evgnanii commented on pom.xml:
```diff
@@ -54,6 +54,23 @@
+            <artifactId>hibernate-core</artifactId>
+            <version>5.6.15.Final</version>
+        </dependency>
+        <!-- Spring ORM для интеграции JPA -->
```
remove comments

Evgnanii commented on src/main/java/com/alex/tasktable/config/AppConfig.java:
```diff
@@ -23,6 +29,29 @@ public BasicDataSource basicDataSourceDataSource() throws IOException {
+
+    private java.util.Properties hibernateProperties() {
+        java.util.Properties properties = new java.util.Properties();
+        properties.put("hibernate.hbm2ddl.auto", "update");
```
user constants for properties

Evgnanii commented on src/main/java/com/alex/tasktable/config/AppConfig.java:
```diff
@@ -23,6 +29,29 @@ public BasicDataSource basicDataSourceDataSource() throws IOException {
+        return sessionFactory;
+    }
+
+    private java.util.Properties hibernateProperties() {
```
java.util.Properties -> just Properties

Evgnanii commented on src/main/java/com/alex/tasktable/model/Task.java:
```diff
@@ -2,14 +2,23 @@
     private Long id;
+    @Column(name = "name", nullable = false)
     private String name;
+    @Column(name = "description", nullable = false)
```
description can be null

Evgnanii commented on src/main/java/com/alex/tasktable/repository/impl/TaskRepositoryImpl.java:
```diff
@@ -2,139 +2,96 @@
+    // Внедрение LocalSessionFactoryBean через конструктор
+    @Autowired
+    public TaskRepositoryImpl(LocalSessionFactoryBean sessionFactoryBean) {
+        this.sessionFactory = sessionFactoryBean.getObject();  // Получаем SessionFactory из LocalSessionFactoryBean
```
can we just use @Autowired SessionFactory ?

Evgnanii commented on src/main/java/com/alex/tasktable/repository/impl/TaskRepositoryImpl.java:
```diff
@@ -2,139 +2,96 @@
-                task.setDeadline(resultSet.getTimestamp("deadline").toLocalDateTime());
-                task.setStatus(resultSet.getString("status"));
+        try (Session session = sessionFactory.openSession()) {
+            Task task = session.get(Task.class, id);
```
use transactions read only mode

Evgnanii commented on src/main/java/com/alex/tasktable/repository/impl/TaskRepositoryImpl.java:
```diff
@@ -2,139 +2,96 @@
+            transaction = session.beginTransaction();
+            Task task = session.get(Task.class, id);
+            if (task == null) {
+                throw new BadRequestException("Task not found with ID: " + id);
```
удаление не должно вызывать ошибку в случае отсутствия ресурса

Evgnanii replied:
ответ всегда 204, DELETE метод идемпотентный

AlexLukashevich95 committed:
Feature-task5:
-change ApplicationException to unchecked
-added yaml constants for hibernate
-delete unnecessary comments
-used session readonly mode

Evgnanii commented on src/main/java/com/alex/tasktable/controller/TaskController.java:
```diff
@@ -11,52 +10,33 @@
-        } catch (ApplicationException e) {
-            throw new RuntimeException(e);
-        }
+        taskdto.setId(id);//чувствую, что надо сетать в jsp, но решил сделать так
```
тут нужно. в дто это поле должно приходить null с  фронта, фронту должно быть по контракту недоступно заполнять это поле

Evgnanii commented on src/main/java/com/alex/tasktable/repository/impl/TaskRepositoryImpl.java:
```diff
@@ -2,139 +2,85 @@
+        try (Session session = sessionFactory.openSession()) {
+            session.setDefaultReadOnly(true);
+            Task task = session.get(Task.class, id);
+            if (task == null) {
```
уверен что нужна тут эта проверка? Хибер вернет null если будет не тот айдилник?

Evgnanii commented on src/main/java/com/alex/tasktable/repository/impl/TaskRepositoryImpl.java:
```diff
@@ -2,139 +2,85 @@
+            session.setDefaultReadOnly(true);
+            Task task = session.get(Task.class, id);
+            if (task == null) {
+                throw new BadRequestException("Task not found with ID: " + id);
```
если эта проверка действительно требуется, сделать чтобы метод репозиторий возвращал Optional<Task>
А в сервисе проверяй уже что в Optional и выбрасывай ошибку если да

AlexLukashevich95 committed:
Feature-task5:
-delete if statement in findid method
